### PR TITLE
The primary version of the postcard generator

### PR DESCRIPTION
**Pull request by DrumovMax**
This project is a postcard generator. The postcard description is entered using voice input. The type of postcard is selected from the drop-down list. The project is written in python, streamlit, hugging face models(stable-diffusion-v1-5, wav2vec2-large-960h-lv60-self). The generation can be run with cuda(with NVIDIA video card) or on CPU.

### CODE REVIEW

**Comment by DrumovMax** [comment]
Reviewer: @SpirinEgor

File: src/postcard_generator.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    try:
+        listener = sr.Recognizer()
+        with sr.Microphone() as microphone:
+            aud_msg = listener.listen(microphone)
+        txt_msg = listener.recognize_google(aud_msg, language='en-US')
+    except Exception as e:
+        st.sidebar.exception(e)
+        st.sidebar.write("We gave you back kitty ʕ ᵔᴥᵔ ʔ")
+        txt_msg = "kitty"
+    return txt_msg.lower()
```
Comment (SpirinEgor) [inline src/postcard_generator.py]: What if I say `kitty`? The app would handle this as an error?
You need to handle exception more accurately, the most simple solution is to return `None`.

Also, showing the user an exception message is a bad idea. In most cases, the user understands nothing. You should handle errors by yourself and show a beautiful message that describes what went wrong and how to fix it.

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    if "postcard" not in st.session_state:
+        default_postcard()
+
+    generate_postcard = st.sidebar.button('Generate postcard')
+    try:
+        if generate_postcard:
+            st.session_state.postcard = get_postcard(st.session_state.record, theme, steps, height, width)
+            st.sidebar.success("Your postcard is ready (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧")
+    except Exception as e:
+        st.sidebar.exception(e)
```
Comment (SpirinEgor) [inline src/main.py]: Same comment about handling exceptions. Don't show them directly to users 😅

File: src/postcard_generator.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    image = pipe(prompt=prompt, num_inference_steps=steps, height=height, width=width).images[0]
+    return image
+
+
+def record_voice():
+    try:
+        listener = sr.Recognizer()
+        with sr.Microphone() as microphone:
+            aud_msg = listener.listen(microphone)
+        txt_msg = listener.recognize_google(aud_msg, language='en-US')
```
Comment (SpirinEgor) [inline src/postcard_generator.py]: Btw, you can find ASR models on HugginFace. So you can combine 2 neural networks in your app 😍
https://huggingface.co/models?pipeline_tag=automatic-speech-recognition

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Wonderful idea, a good example of how to use NN 👍
Provide a better solution for working with exceptions, please.

File: src/postcard_generator.py (hunk 1 above)
Comment (DrumovMax) [reply src/postcard_generator.py]: Done!

File: src/main.py (hunk 2 above)
Comment (DrumovMax) [reply src/main.py]: Changed!

File: src/postcard_generator.py (hunk 3 above)
Comment (DrumovMax) [reply src/postcard_generator.py]: Done!

**Comment by DrumovMax** [comment]
Made changes to the implementation of the application. @SpirinEgor

**Review by SpirinEgor — APPROVED** [review]
Very cool idea 👍

**Pull request merged**